test/test_client.js

```diff
@@ -1,66 +1,91 @@
+    context('with no api key at all', function(){
+        before(function() {
             delete process.env['TWINGLY_SEARCH_KEY'];
-            var c = new Client();
-        }).to.throw('No API key has been provided.');
+        });
+
+        after(function() {
+            process.env['TWINGLY_SEARCH_KEY'] = setup.randomValueHex(16);
+        });
+
+        it('should throw error', function(){
+            expect(function(){
+                var c = new Client();
+            }).to.throw('No API key has been provided.');
+        });
     });
 
-    it('with optional user agent given', function(){
-        var c = new Client(null, 'Test User-Agent');
-        expect(c.userAgent).to.be.equal('Test User-Agent');
+    context('with optional user agent given', function(){
+        it('should use that user agent', function(){
+            var c = new Client(null, 'Test User-Agent');
+            expect(c.userAgent).to.be.equal('Test User-Agent');
+        });
     });
 
-    it('query', function(){
-        var q = (new Client()).query();
-        expect(q).to.be.instanceof(Query);
+    describe('#query', function(){
+        it('should return an instance of Query', function(){
+            var q = (new Client()).query();
+            expect(q).to.be.instanceof(Query);
+        });
     });
 
-    it('execute query with invalid api key', function(done){
-        process.env['TWINGLY_SEARCH_KEY_TEMP'] = process.env['TWINGLY_SEARCH_KEY'];
-        process.env['TWINGLY_SEARCH_KEY'] = 'wrong';
+    describe('#execute_query', function(done){
+        context('with invalid api key', function(){
+            before(function() {
+                process.env['TWINGLY_SEARCH_KEY'] = 'wrong';
+            });
 
-        nvcr.insertCassette('search_without_valid_api_key');
-        var c = new Client();
-        var q = c.query();
-        q.pattern = 'something';
-        c.executeQuery(q, function(error, result){
-            nvcr.ejectCassette();
-            expect(error).to.be.instanceof(TwinglyAuthError);
-            done();
+            after(function() {
+                process.env['TWINGLY_SEARCH_KEY'] = setup.randomValueHex(16);
+            });
+
+            it('should throw error on invalid API key', function() {
+                nvcr.insertCassette('search_without_valid_api_key');
+                var c = new Client();
+                var q = c.query();
+                q.pattern = 'something';
+                c.executeQuery(q, function(error, result){
+                    nvcr.ejectCassette();
+                    expect(error).to.be.instanceof(TwinglyAuthError);
+                    done();
+                });
+            });
         });
     });
 
-    it('endpoint url', function(){
+    describe('#endpoint_url', function(){
         var c = new Client();
-        expect(c.endpointUrl()).to.be.equal(c.BASE_URL + c.SEARCH_PATH)
+        var expected = c.BASE_URL + c.SEARCH_PATH;
+
+        it('should eq '+expected, function() {
```

Comment:
Nope, just didn't think about it

we also mix `function() {` and `function(){`, I think we should address all these style issues separately